packages/dds/sequence/package.json

```diff
@@ -194,6 +194,9 @@
 		"broken": {
 			"InterfaceDeclaration_SequenceOptions": {
 				"backCompat": false
+			},
+			"VariableDeclaration_SharedString": {
```

Comment:
These showing up as broken seems like a defect in our type testing. Adding a member to the type of a variable (especially when its type is sealed) should be non-breaking. I think adjusting our type tests to consider all variable types (cases using typeof) sealed might be a good idea. That would avoid classifying this as a breaking change, which seems like it would be an improvement. Thoughts?

Reply:
ehh I think there are cases where we legitimately need to be careful doing things like this because we claim to support some limited configurations of cross-layer multi-versioned support. E.g. suspend disbelief in which bits of our system depend on specific DDSes, and say container-runtime depended on SharedString. It would be problematic to make this change if we claimed support for container-runtime version N+1 (which started using the new API I added) with data-runtime N (which doesn't have this API).

This change is obviously not such a worrisome case, but having noisier type tests *can* be a useful signal for devs.

We could definitely be doing better here, but I want whatever effort happens to consider cases like the above and not assume things are OK strictly because an interface is sealed.

Reply:
Our type tests already assume adding things to sealed interfaces is ok (perhaps that should be revisited, but thats already in and not want I'm discussing). What they are not assuming is that adding members to variables (even if their type is sealed is a sealed interface) is ok. This is why you got 8 broken APIs for each of the ShareObjectKind concreate instance, and none for the interface modification.

We also don't have any checks for any changes to static members of classes at all. Generally our handling of type tests for the `typeof` cases are not great in my opinion. I'll make a proposed improvement for that in a bit.

Anyway, I don't think type tests are the correct place to enforce compatibility requirements that are not captured by semver (at least without opting into extra type testing to cover it). We reset the type test baselines every release, and consider breaks in them ok across majors. This makes them very unsuited that 1.x works with 2.x in the very specific ways that mixed versioning requires. Perhaps packages which end up with these extra strict requirements  could have configuration (or annotations on the relevant types) to opt into additional type tests, but I don't think erroring on adding members to all variables is helpful for those cases (I don't think the relevant packages have many variables: they are mostly just interfaces), and I think reducing false positive noise in type tests would really help increase scrutiny on cases they actually detect. I don't know the details of that aspect of our compat very well though, so I may be missing something.

Reply:
Type test change, discussion can move over there: https://github.com/microsoft/FluidFramework/pull/21876